File: accommodation_service/src/main/java/huy/project/accommodation_service/infrastructure/repository/adapter/LocationAdapter.java
package huy.project.accommodation_service.infrastructure.repository.adapter;

import huy.project.accommodation_service.core.domain.entity.LocationEntity;
import huy.project.accommodation_service.core.port.ILocationPort;
import huy.project.accommodation_service.infrastructure.repository.ILocationRepository;
import huy.project.accommodation_service.infrastructure.repository.mapper.LocationMapper;
import huy.project.accommodation_service.infrastructure.repository.model.LocationModel;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class LocationAdapter implements ILocationPort {
    private final ILocationRepository locationRepository;

    @Override
    public LocationEntity save(LocationEntity location) {
        LocationModel locationModel = LocationMapper.INSTANCE.toModel(location);
        return LocationMapper.INSTANCE.toEntity(locationRepository.save(locationModel));
    }

    @Override
    public LocationEntity getLocationById(Long id) {
        return locationRepository.findById(id).map(LocationMapper.INSTANCE::toEntity).orElse(null);
    }

    @Override
    public void deleteLocationById(Long id) {

    }
}

File: location_service/LocationService.Infrastructure/Repository/Adapter/LocationAdapter.cs
using LocationService.Core.Domain.Entity;
using LocationService.Core.Port;
using LocationService.Infrastructure.Repository.Mapper;
using Microsoft.EntityFrameworkCore;

namespace LocationService.Infrastructure.Repository.Adapter
{
    public class LocationAdapter : ILocationPort
    {
        private readonly LocationDbContext _dbContext;

        public LocationAdapter(LocationDbContext dbContext)
        {
            _dbContext = dbContext;
        }

        public async Task<LocationEntity> CreateLocationAsync(LocationEntity location)
        {
            var model = LocationMapper.ToModel(location);
            _dbContext.Locations.Add(model);
            await _dbContext.SaveChangesAsync();
            return LocationMapper.ToEntity(model);
        }

        public async Task<LocationEntity> GetLocationByIdAsync(long id)
        {
            var model = await _dbContext.Locations.FindAsync(id);
            return LocationMapper.ToEntity(model);
        }

        public async Task<List<LocationEntity>> GetLocationsByIdsAsync(List<long> ids)
        {
            var models = await _dbContext.Locations
                .Where(l => ids.Contains(l.Id))
                .ToListAsync();
            return models.Select(LocationMapper.ToEntity).ToList();
        }
    }
}